src/main/webapp/app/modules/canvas/visualization/canvas.tsx

```diff
@@ -143,10 +97,49 @@ const Canvas = (props: VisualizationProp) => {
     }
   }, []);
 
+  useEffect(() => {
+    if (props.isSocketConnected) {
+      VisualMetadataContainerAdd(props.visualmetadata?.visualMetadataSet);
+      loadVisualization();
+    }
+  }, [props.isSocketConnected]);
+
+  useEffect(() => {
+    if (props.visualData) {
+      if (props.visualData?.headers?.request === 'filters') {
+        const obj = props.visualData?.body[0];
+        let dimensionName = '';
+        for (const i in obj) {
+          dimensionName = i;
+          break;
+        }
+        const retVal = props.visualData?.body?.map(function (item) {
+          return {
+            value: item[dimensionName],
+            label: item[dimensionName],
+          };
+        });
+        props.filterData[dimensionName] = retVal;
+      } else {
+        const v = VisualMetadataContainerGetOne(props.visualData.headers.queryId);
+        if (v && props.visualData?.body.length > 0) {
+          v.data = props.visualData?.body;
+          $(`.loader-${v.id}`).hide();
+          $(`.dataNotFound-${v.id}`).hide();
+          renderVisualization(v, props.visualData?.body);
+        } else {
+          $(`.dataNotFound-${v.id}`).show();
+          $(`.loader-${v.id}`).hide();
```

Comment:
you only required jquery for visualizations..use redux reducer state to show and hide...this is not accepted

Reply:
here I want to hide or show div base id

id is selected visualization
how I can use state here

Reply:
store viz related information using redux as flag and map with component then use ? opt to show hide particular html element